File: Python/Ex 23.py
#ex23
v1 = float(input("Digite o primeiro valor: "))
v2 = float(input("Digite o segudo valor: "))
if v1>v2:
  print("O segundo número menor que o primeiro.")
else:
  v3 = float(input("Digite o terceiro valor: "))
  if v3<v2:
    print("O terceiro número é menor que o segundo")
  else:
    v4 = float(input("Digite o quarto valor: "))
    if v4>=v3:
      print(v4, v3, v2, v1)
    else:
      if v4>=v2:
        print(v3, v4, v2, v1)
      else:
        if v4>=v1:
          print(v3, v2, v4, v1)
        else:
          print(v3, v2, v1, v4)

File: Java/Exercicios, vetores e matrizes/ex23.java
/*
    23.	Receba 3 valores obrigatoriamente em ordem crescente e um 4º valor não necessariamente em ordem. Mostre os 4 números em ordem crescente.
*/
import javax.swing.JOptionPane;
public class ex23 {
    static double n1, n2, n3, n4;
    public static void main(String args[]){
        ReceberValores();
        }
    
    static void ReceberValores(){
        n1 = Double.parseDouble(JOptionPane.showInputDialog("Digite o primeiro valor: "));
        n2 = Double.parseDouble(JOptionPane.showInputDialog("Digite o segundo valor: "));
        if (n2<n1){
            JOptionPane.showMessageDialog(null, "O número digitado não pode ser menor que o anterior.");
            }
        else{
            n3 = Double.parseDouble(JOptionPane.showInputDialog("Digite o terceiro valor: "));
            if (n3<n2){
                JOptionPane.showMessageDialog(null, "O número digitado não pode ser menor que o anterior.");
                }
            n4 = Double.parseDouble(JOptionPane.showInputDialog("Digite o quarto número: "));
            MostraValores();
            }
    }
    
    static void MostraValores(){
        if (n4<n1){
            JOptionPane.showMessageDialog(null, "Os números digitados em ordem crescente são: " +n4 +", " +n1 +", " +n2 +", " +n3);
        }
            else{
                if(n4<n2 && n4>n1){
                    JOptionPane.showMessageDialog(null, "Os números digitados em ordem crescente são: " +n1+", " +n4 +", " +n2 +", " +n3);
                }
                else{
                    if(n4<n3 && n4>n2){
                        JOptionPane.showMessageDialog(null, "Os números digitados em ordem crescente são: " +n1 +", " +n2 +", " +n4 +", " +n3);
                    }
                    else{
                        JOptionPane.showMessageDialog(null, "Os números digitados em ordem crescente são: " +n1 +", " +n2 +", " +n3 +", " +n4);
                    }
                }
            }
    }
}
